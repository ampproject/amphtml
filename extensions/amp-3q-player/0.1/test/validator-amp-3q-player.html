<!--
  Test Description:
  Tests for the amp-3q-player tag. See the inline comments.
-->
<!DOCTYPE html>
<html amp lang="en">

<head>
  <meta charset="utf-8">
  <title>3Q AMP Example</title>
  <link rel="canonical" href="3q-player.amp.html">
  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
  <style amp-boilerplate>
    body {
      -webkit-animation: -amp-start 8s steps(1, end) 0s 1 normal both;
      -moz-animation: -amp-start 8s steps(1, end) 0s 1 normal both;
      -ms-animation: -amp-start 8s steps(1, end) 0s 1 normal both;
      animation: -amp-start 8s steps(1, end) 0s 1 normal both
    }

    @-webkit-keyframes -amp-start {
      from {
        visibility: hidden
      }

      to {
        visibility: visible
      }
    }

    @-moz-keyframes -amp-start {
      from {
        visibility: hidden
      }

      to {
        visibility: visible
      }
    }

    @-ms-keyframes -amp-start {
      from {
        visibility: hidden
      }

      to {
        visibility: visible
      }
    }

    @-o-keyframes -amp-start {
      from {
        visibility: hidden
      }

      to {
        visibility: visible
      }
    }

    @keyframes -amp-start {
      from {
        visibility: hidden
      }

      to {
        visibility: visible
      }
    }
  </style>
  <noscript>
    <style amp-boilerplate>
      body {
        -webkit-animation: none;
        -moz-animation: none;
        -ms-animation: none;
        animation: none
      }
    </style>
  </noscript>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-3q-player" src="https://cdn.ampproject.org/v0/amp-3q-player-latest.js"></script>
</head>

<body>

  <!-- valid -->
  <amp-3q-player data-id="c8dbe7f4-7f7f-11e6-a407-0cc47a188158" height=360 width=640 layout="responsive">
  </amp-3q-player>

  <!-- invalid, needs data-id -->
  <amp-3q-player height=360 width=640 layout="responsive" data-id="">
  </amp-3q-player>

  <h2>Data player</h2>
  <amp-3q-player height=360 width=640 layout="responsive" data-id="" data-player="c8dbe7f4-7f7f-11e6-a407-0cc47a188158">
  </amp-3q-player>

  <h2>Resize</h2>
  <amp-3q-player id="myVideo" data-id="7e3fc527-a9b5-11eb-b839-0cc47a188158" layout="responsive" height="110"
    width="640">
  </amp-3q-player>

</body>

</html>