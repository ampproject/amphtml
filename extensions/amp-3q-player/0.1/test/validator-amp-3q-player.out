FAIL
|  <!--
|    Copyright 2017 The AMP HTML Authors. All Rights Reserved.
|
|    Licensed under the Apache License, Version 2.0 (the "License");
|    you may not use this file except in compliance with the License.
|    You may obtain a copy of the License at
|
|        http://www.apache.org/licenses/LICENSE-2.0
|
|    Unless required by applicable law or agreed to in writing, software
|    distributed under the License is distributed on an "AS-IS" BASIS,
|    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
|    See the License for the specific language governing permissions and
|    limitations under the license.
|  -->
|  <!--
|    Test Description:
|    Tests for the amp-3q-player tag. See the inline comments.
|  -->
|  <!DOCTYPE html>
|  <html amp lang="en">
|    <head>
|      <meta charset="utf-8">
|      <link rel="canonical" href="./regular-html-version.html">
|      <meta name="viewport" content="width=device-width,minimum-scale=1">
|      <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
|      <script async custom-element="amp-3q-player" src="https://cdn.ampproject.org/v0/amp-3q-player-latest.js"></script>
|      <script async src="https://cdn.ampproject.org/v0.js"></script>
|    </head>
|    <body>
|
|      <!-- valid -->
|      <amp-3q-player
|        data-id="c8dbe7f4-7f7f-11e6-a407-0cc47a188158"
|        height=360
|        width=640
|        layout="responsive">
|      </amp-3q-player>
|
|      <!-- invalid, needs data-id -->
|      <amp-3q-player
>>     ^~~~~~~~~
amp-3q-player/0.1/test/validator-amp-3q-player.html:41:4 The mandatory attribute 'data-id' is missing in tag 'amp-3q-player'. (see https://amp.dev/documentation/components/amp-3q-player)
|        height=360
|        width=640
|        layout="responsive">
|      </amp-3q-player>
|    </body>
|  </html>
